typescript/api/services/VocabService.ts

```diff
@@ -146,6 +149,112 @@ export module Services {
       return response.data;
     }
 
+    public async findRecords(sourceType:string, brand:BrandingModel, searchString:string, start:number, rows:number): Promise<any> {
+
+      const report = sails.config.vocab[sourceType];
```

Comment:
Rather than using the report for the variable name, query seems more appropriate.

Create a config model in core for this query object and set the query object to this type.

Reply:
Should also be using sails.config.vocab.queries